src/CentralLimitGraph.jsx

```diff
@@ -229,35 +230,47 @@ export class CentralLimitGraph extends Component {
         const population = this.generatePopulation(
             key === 'populationSize' ? value : this.state.populationSize,
             key === 'mean' ? value : this.state.mean,
-            key === 'stdDev' ? value : this.state.stdDev);
+            key === 'stdDev' ? value : this.state.stdDev,
+            key === 'seed' ? value : this.state.seed
+        );
         const populationGraphData = createHistogramArray(population);
         this.setState({
             population: population,
             populationGraphData: populationGraphData,
             [key]: value
         });
     }
-    generatePopulation(size, mean, stdDev) {
+    generatePopulation(size, mean, stdDev, seed) {
+        // Reset the global Math.random everytime this is called
+        seedrandom(seed, {global: true});
         return jStat.create(1, size, (row) => {
             // check this
             row;
             let i = jStat.normal.sample(mean, stdDev);
-            return roundFloat(i);
+            return math.round(i, 1);
         })[0];
     }
     runSample() {
+        // Use the base64 encoding of the seed as a simple hash
+        let samplingSeed = window.btoa(this.state.seed);
+        let ng = seedrandom(samplingSeed);
+
         // take samples from population values
         // - one sample consists of N sampleSize, which are then averaged
         // - N numberOfSamples are taken
         // push these to sampleSet
         // get the histogram and set state, render a line graph
-        let sampleMeans = [];
+        let sampleMeans = new Array(this.state.numberOfSamples);
         for (var i = 0; i < this.state.numberOfSamples; i++) {
-            let samples = _.sampleSize(
-                this.state.population,
-                this.state.sampleSize);
-            sampleMeans.push(roundFloat(_.mean(samples)));
+            let samples = new Array(this.state.sampleSize);
+            for (var j = 0; j < this.state.sampleSize; j++) {
+                let observationIdx = Math.floor(ng() * this.state.population.length);
+                samples[j] = this.state.population[observationIdx];
```

Comment:
Some sources online report that its faster to index into an array rather than `.push()`.  Of course this all depends on the browser so there's no single answer.

For whoever reviews this, which is more clear: indexing or pushing?